pkg/k8s/client.go

```diff
@@ -28,12 +29,13 @@ import (
 )
 
 // NewClient returns a new kubernetes.Controller which means to provide access to locally-cached k8s resources
-func NewClient(osmNamespace, meshConfigName string, informerCollection *osminformers.InformerCollection, policyClient policyv1alpha1Client.Interface, msgBroker *messaging.Broker, selectInformers ...InformerKey) *Client {
+func NewClient(osmNamespace, meshConfigName string, informerCollection *osminformers.InformerCollection, kubeClient kubernetes.Interface, policyClient policyv1alpha1Client.Interface, msgBroker *messaging.Broker, selectInformers ...InformerKey) *Client {
```

Comment:
Is this the right object/place to be referencing the Kubeclient in? Still getting use to the various interfaces/classes used to interact with k8s.

Reply:
based on https://github.com/openservicemesh/osm/pull/5018 it looks like k8s client is for CRUD, so this would be the place to add ability to update secrets.

Reply:
I was surprised since we didn't have it in this pkg/k8s/client but looks like it was primarily informers in this package and pass the k8s client around when needed but in process of simplifying it 👍